### ISSUE TITLE
No option to specify RAM amount in "Create a new qube" window

### ISSUE THREAD

**Issue by TNTBOMBOM** [QubesOS/qubes-issues#5198]
**Qubes OS version**
Qubes 4.0.1

**Affected component(s) or functionality**
Booting Empty standalone qube

**Brief summary**
Cant boot my debian.iso through a standalone qube PVH , HVM

**To Reproduce**
_Steps to reproduce the behavior:
_1. Creat Empty standalone qube
_2. go to settings and choose to boot qube from CDROM
_3. choose where the .iso is located
_4. The standalone qube will boot for couple of seconds then shutdown itself 

**Expected behavior**
To continue and boot from the .iso image

**Actual behavior**
the qube is not booting , always shutting down

**Screenshots**
none

**Additional context**

I went as well to the settings of the standalone empty vm and changed it from PVH to HVM. Same issue.

**Solutions you've tried**

Instead of creating an empty qube and go to settings and from there booting the CD , i open the .iso while i want to create the empty qube which is: 

![qubesempty1](https://user-images.githubusercontent.com/11895339/62000570-a2ef0800-b0c9-11e9-8302-d341c2515539.png)

then choosing the iso image , and guess what it will boot. 

BUT There are two issues in here:

- The ram which is the default 400mg cant make the OS go through the installation and the os will lag if you proceed with graphical installation. unless you switch it to the non-graphical installation. (so this is issue needs to be fixed which is the ability to change ram capacity before booting the iso, or increasing ram to pvh/hvm by default , or fixing the ability of consuming rams through ram balance).

- While installing the OS (debian im using) , it says to enter my netmask 
default netmask in the os recommendation: 255.255.255.0
Qube netmask: 255.255.255.255

so i switched the netmask to .255 

- The gateway suggested in the OS is 10.137.0.15 , while the gateway IP in qube is 10.137.0.5 changing the gateway IP in the OS to .5 instead of .15 will result in: 

![qubesempty2](https://user-images.githubusercontent.com/11895339/62001163-527fa680-b0d9-11e9-91d1-261c686c6cc6.png)

![qubesempty3](https://user-images.githubusercontent.com/11895339/62001157-409e0380-b0d9-11e9-952f-a19b87d360f2.png)

Note: i even switched the gateway & netmask according to OS recommendation in order to proceed then changing that in the OS results are the same with no internet connection.

**Comment by unman** [comment]
If you go to Settings, then it is your responsibility to make sure you have the configuration right.
1. RAM - this is covered at https://www.qubes-os.org/doc/hvm/#creating-an-hvm-qube

Specifically:

> Make sure that you give the new qube adequate memory to install and run.

Qubes has no way of knowing how much RAM your OS will need - BSDs can install in tiny (default) amounts, some versions of Windows and Kali seem to need a lot. Qubes cant predict what you will need, and you have the ability to set it yourself.
(Your first option - " the ability to change ram capacity before booting the iso" - is already there.)

2. Netmask - duplicate of #4189.
If you use 255.255.255.255 then your qube has no idea how to access any other IP address unless you give specific routes.
I don't know where the .15 address came from, but you should be able to see it is obviously wrong. You can set Qubes gateway (.5) and mask 255.255.255.0 if you want networking.

**Comment by TNTBOMBOM** [comment]
> If you go to Settings

Im talking about this part:

![qubesempty1](https://user-images.githubusercontent.com/11895339/62010960-5525de80-b161-11e9-9cbb-818845507ca1.png)

Means no way to increase/decrease rams , better to be added as well to the screen. Going to settings means some options on that page are useless.

> I don't know where the .15 address came from, but you should be able to see it is obviously wrong.

![qubesnontemplate1](https://user-images.githubusercontent.com/11895339/62011013-3b38cb80-b162-11e9-8259-eca3b8d4feef.png)

Talking about his one. 

> You can set Qubes gateway (.5) and mask 255.255.255.0 if you want networking

The key is changing netmask to 255.255.255.0 not using the same netmask of qube and the installation will be proceeded. Thank you!

On the other hand there are as several issue from above bug: 

- PVH = qube not booting with debian buster .iso
- Switching PVH to HVM from settings of the same qube = same issue which is not booting
- Using this feature: 
![qubesempty1](https://user-images.githubusercontent.com/11895339/62011071-00836300-b163-11e9-8f89-52f687470954.png)
It will load the .iso image but it has no settings for RAMs. It seems as well it creates HVM at the first shot thus allowing the .iso to load. 
- Netmask of qube = good thing has been reported before.

**Comment by andrewdavidwong** [comment]
If I understand correctly, the bug for this issue is the lack of an option to specify a RAM amount when creating a new qube where the OS for that qube is installed from a device.

If this is incorrect, please let me know.

If this is correct but there are also other, separate bugs, please file separate issues for them.

**Comment by marmarta** [comment]
I think the issue is actually lack in documentation - if we had in docs some sample "how to create HVM with this-and-that-system" procedures using GUI, it would be easier. Not that big a fan of adding the option to Create New VM window - there's an option to launch settings there already.

**Comment by TNTBOMBOM** [comment]
@marmarta isnt that page "[dom0] Create new qube" missing the option? because what is the point of showing that page if the user need to untick "install system from.." which is the default ticket option for users.

**Comment by marmarek** [comment]
There is a "launch settings after creation" option if you want to change anything before starting it. It will also automatically uncheck the "install system from device (also available from settings)" option. 
So, basically you have a choice:
 - start a qube with default settings directly
 - open settings to modify something and start installation from there

Isn't that clear enough? Would you like those option to be rephrased somehow?

**Comment by TNTBOMBOM** [comment]
> There is a "launch settings after creation" option if you want to change anything before starting it.

True but its not booting the .iso image unless i choose "install system from device..." from "[dom0] Create new qube"

> Isn't that clear enough?

Clear for advanced or at least user who got used to Qubes, not for new comers. 

>  Would you like those option to be rephrased somehow?

Possible as well like add: 

`Launch settings after creation (useful to change RAMs or Storage capacity...)`

Maybe this is useful to give clear hint about the purpose of it, thus its not ideal solution similarly as adding these options to the qube creating page itself.

**Comment by TNTBOMBOM** [comment]
#5203 resolved with missed package (zenity) which hopefully can be added to the documentation. 

So i leave this suggestion/enhancement to add RAM,Storage capacity options to you to decide. I see it very useful to have all things in one place instead of creating then going then missing going there...etc (specially for new comers).
